### PR TITLE
feat: add more feilds to send for API

### PR DESCRIPTION
**Pull request by tejaskh3**
Date: 25 April. 2024

Developer Name: Tejas

---

## Description
Adding some more fields to **<discorBaseUrl>/task/update** API to send task's updates on RDS's discord channel.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
image

</details>

## Test Coverage
image

image

image

</details>

### CODE REVIEW

File: src/utils/sendTaskUpdates.ts [hunk 1]
```diff
@@ -5,9 +5,17 @@ export async function sendTaskUpdate(
   completed: string,
   planned: string,
   blockers: string,
+  discordId: string,
+  taskId: string,
   env: env
 ): Promise<void> {
-  const formattedString = `**Completed**: ${completed}\n\n**Planned**: ${planned}\n\n**Blockers**: ${blockers}`;
+  const taskUrl = "https://status.realdevsquad.com/tasks/" + taskId;
```
Comment (tejaskh3) [inline src/utils/sendTaskUpdates.ts:12]: Should I keep this hard-coded here?
It is just to provide the link clickable like of task.

**Review by ardourApeX — CHANGES_REQUESTED** [review]
As far as I remember Ankush said no need to tag the user while posting an update. Also, the task link should be hyperlinked.

**Comment by tejaskh3** [comment]
> As far as I remember Ankush said no need to tag the user while posting an update. Also, the task link should be hyperlinked.

Yes the task link is a hyperlink, and about taging the user I tried to find a solution where we can print userName without taging user but not able to find something for that.

File: src/utils/sendTaskUpdates.ts (hunk 1 above)
Comment (tejaskh3) [reply src/utils/sendTaskUpdates.ts:12]: Thank you for your clarification

File: src/utils/sendTaskUpdates.ts [hunk 2]
```diff
@@ -5,9 +5,17 @@ export async function sendTaskUpdate(
...
+  const formattedString =
+    `<@${discordId}> added an update to his task: <${taskUrl}>\n` +
+    `**Completed**\n${completed}\n\n` +
+    `**Planned**\n${planned}\n\n` +
+    `**Blockers**\n${blockers}`;
+  console.log(formattedString);
```
Comment (lakshayman) [inline src/utils/sendTaskUpdates.ts:18]: I think we should remove this console.

**Review by lakshayman — CHANGES_REQUESTED** [review]

File: src/utils/sendTaskUpdates.ts [hunk 3]
```diff
@@ -5,9 +5,17 @@ export async function sendTaskUpdate(
...
   planned: string,
   blockers: string,
+  discordId: string,
+  taskId: string,
   env: env
 ): Promise<void> {
-  const formattedString = `**Completed**: ${completed}\n\n**Planned**: ${planned}\n\n**Blockers**: ${blockers}`;
+  const taskUrl = "https://status.realdevsquad.com/tasks/" + taskId;
+  const formattedString =
+    `<@${discordId}> added an update to his task: <${taskUrl}>\n` +
```
Comment (lakshayman) [inline src/utils/sendTaskUpdates.ts:14]: We should use a gender-neutral message. Like `They added an update to their task`.

File: src/utils/sendTaskUpdates.ts (hunk 1 above)
Comment (tejaskh3) [reply src/utils/sendTaskUpdates.ts]: added please check

File: src/utils/sendTaskUpdates.ts (hunk 2 above)
Comment (tejaskh3) [reply src/utils/sendTaskUpdates.ts]: Done please check

File: src/utils/sendTaskUpdates.ts (hunk 3 above)
Comment (tejaskh3) [reply src/utils/sendTaskUpdates.ts]: Done please check

**Comment by tejaskh3** [comment]
> > As far as I remember Ankush said no need to tag the user while posting an update. Also, the task link should be hyperlinked.
> 
> I tried to find a solution where we can print userName without tagging user but was not able to find something for that, and about.

Done now, we are not tagging user.

**Comment by ankushdharkar** [comment]
@tejaskh3 For now, we can display the task link instead of title. But once this feature is live, please work on https://github.com/Real-Dev-Squad/website-backend/issues/2025

**Review by lakshayman — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**